### ISSUE TITLE
Tabs a11y: Need way to pass element to right of tablist while keeping outside of role="tablist" element.

### ISSUE THREAD

**Issue by bvandrc** [palantir/blueprint#6217]
#### Feature request

Tabs: Need way to pass element to right of tablist while keeping outside of `role="tablist"` element.

For example, on the docs app, this searchbar:

![image](https://github.com/palantir/blueprint/assets/87083504/eceac06a-346d-4226-a3ad-e44f3bc0173b)

Is currently created like such:

![image](https://github.com/palantir/blueprint/assets/87083504/458ab925-32dc-4fa7-b1e4-4f2ca9d43d8f)

But the issue with this is that the `input` element gets put within the element that has `role="tablist"`. This is a major accessibility failure, as this input element is not truly part of the list of tabs; it's a separate component that is adjacent to the list of tabs. 

Desired implementation would be something like:

![image](https://github.com/palantir/blueprint/assets/87083504/0281d30f-4ee9-4edf-9621-968ea1c1ae47)

With the expected result being visually the same.

**Comment by adidahiya** [comment]
> it's a separate component that is adjacent to the list of tabs.

Then it sounds like you should probably render it as a sibling of `<Tabs>` rather than as a child node. You should be able to do this with a bit of custom styling and rendering your tab panels separately from the tablist. I made a code sandbox to demonstrate this approach: https://codesandbox.io/p/sandbox/blueprint-v5-x-sandbox-react-16-forked-sntdc8?file=%2Fpackage.json%3A10%2C5

![image](https://github.com/palantir/blueprint/assets/723999/088e71eb-bd87-4be3-a695-22cf98522c0b)

**Comment by bvandrc** [comment]
@adidahiya Those examples are not equivalent because even though they look similar, you have not included the tab panel in them. The problem with making it a sibling is that it moves the entire `Tabs` element over:

![image](https://github.com/palantir/blueprint/assets/87083504/d191bb09-795e-49b8-bfc7-201281b6f5c0)

**Comment by adidahiya** [comment]
@bvandercar-vt you're right. I updated my [sandbox](https://codesandbox.io/p/sandbox/blueprint-v5-x-sandbox-react-16-forked-sntdc8?file=%2Fnode_modules%2F%40blueprintjs%2Fcore%2Flib%2Fesm%2Fcomponents%2Ftabs%2Ftab.d.ts%3A6%2C13) to show how it could work with the panels rendered independently of the `<Tabs>` component. You would need to render some of the a11y markup of `.bp5-tab-panel` yourself since Blueprint does not export a `<TabPanel>` component (this is something I would be open to implementing or accepting as a contribution PR). But it is doable:

![image](https://github.com/palantir/blueprint/assets/723999/bd6133f5-0eb5-43b6-9cca-952ffcb63f61)
